pup-0003.md

```diff
@@ -0,0 +1,157 @@
+2.13-dev, the issue's release field should be set to 2.13.next. Then when the
+change is released, it's field will be updated with the actual value (e.g.
+2.13.4). For a hotfix that is applied to a release branch, the release branch is
+set to the hotfix release version.
+
+### Timeframe
+
+When this proposal is accepted, we'll work with release engineering to determine
+a date to switch to cherry-picking. After a date has been picked, an
+announcement will be sent out to the pulp-dev list with the planned date. After
+tihs date, we should begin to follow this proposal. Also, we'll enter in Redmine
+issues to update our docs according after this proposal has been accepted.
+
+## Drawbacks
+
+Git is really built for changes to be merged between branches. For example,
+when cherry-picking a change, the commit is assigned a new id but when merging
+the commit, it retains the same id. It makes looking up the commit to see what
+branches it's been applied to easier. If we cherry-pick, we have to track this
+information externally to git. This means also having to remember to update
+Redmine.
+
+Switching to this model also creates some inconsistency. For example, if we
+adopt this for Pulp 3, then we still have to use merge forwarding for Pulp 2.
+Then we are stuck having to support two different strategies.
+
+Also, when creating bug fixes in the cherry-pick model, code will be developed
+against master. This could cause problems when the code is cherry picked to
+devel branches such as the change relying on code that is in master but not
+in the devel branch. This requires extra scrutiny and awareness on the part
+of the person cherry-picking a change back, to ensure that all code touched or
+used by a change exists on the devel branch and behaves the same. Sometimes
+it will require cherry-picking intermediate changes that might not otherwise be
+obvious.
+
+Defaulting to master, and requiring extra steps to get a change onto the
+current devel branch, is likely to result in fewer bug fixes making it onto
+devel branches. We can and should try to be diligent about cherry-picking fixes
+to the devel branch, but the path of least resistance may prove too tempting
+at times. That said, we could have a schedule (e.g. weekly) where one would
+cherry-pick the changes backward regularly.
+
+## Alternatives
+
+### Keeping the current model
+
+See the prior sections like "Motivation" and "Drawbacks" for a discussion on this.
+
+### Merge forward less often
+
+Perhaps keep the current model but merge less often. This would help to prevent
+accidents and keep the git history cleaner. There are some drawbacks however to
+this. One would be an increased chance of merge conflicts if we merge less
+often. However, we could also use Github's PR functionality to allow for code
+reviews of merges. This would also help to decrease accidental merges.
+
+## Footnotes
+
+[1]: https://docs.pulpproject.org/en/2.12/dev-guide/contributing/branching.html
```

Comment:
In the rendered doc the footnotes section is empty. I really like how these link, but maybe we could add another line so this shows up also in the footnotes section?